lib/open_mic.rb

```diff
@@ -0,0 +1,35 @@
+class OpenMic
+attr_reader :location,
+            :date,
+            :performers
+
+  def initialize(information)
+    @location = information[:location]
+    @date = information[:date]
+    @performers = []
+  end
+
+  def welcome(user)
+    performers << user
+  end
+
+  def jokes
+    all_jokes = []
+    @performers.each do |performer|
+      all_jokes << performer.jokes
+    end.flatten
+    all_jokes
+  end
+
+  def repeated_jokes?
```

Comment:
This is really close, and you were certainly on the right track to solving this problem. I love that you used your `jokes` method to do some of the work. Which enumerable would help us return a new array that is not joke objects anymore, but just joke ids? How can we check for duplicates in a ruby array? (I bet StackOverflow has a few implementations that could get us what we want)